swayfmt/src/formatter/mod.rs

```diff
@@ -1375,6 +1375,115 @@ fn foo() {}
 #[test]
 #[inline(always)]
 fn foo() {}
+"#;
+        let mut formatter = Formatter::default();
+        let formatted_sway_code =
+            Formatter::format(&mut formatter, Arc::from(sway_code_to_format), None).unwrap();
+        assert_eq!(correct_sway_code, formatted_sway_code);
+        assert!(test_stability(formatted_sway_code, formatter));
+    }
+
+    #[test]
+    fn test_comment_between_if_else_overindented() {
+        let sway_code_to_format = r#"contract;
+
+impl MyContract for Contract {
+    fn is_blue() -> bool {
+        if self == PrimaryColor::Blue {
+            true
+        }
+        // TODO remove this else when exhaustive ifs are checked for
```

Comment:
It seems like the new logic is assuming that the comment is correctly aligned. Do we have a test case where both the comment and the else block are misaligned? Something like:

```
impl MyContract for Contract {
    fn is_blue() -> bool {
        if self == PrimaryColor::Blue {
            true
        }
                      // TODO remove this else when exhaustive ifs are checked for
                else {
            false
        }
    }
}"#;
```

Reply:
Can confirm that this is still an issue, output:

```rust
impl MyContract for Contract {
    fn is_blue() -> bool {
        if self == PrimaryColor::Blue {
            true
        }
                // TODO remove this else when exhaustive ifs are checked for
                else {
            false
        }
    }
}"#;
```